lib/signs/utilities/messages.ex

```diff
@@ -258,6 +267,42 @@ defmodule Signs.Utilities.Messages do
     end
   end
 
+  defp overnight_period?(current_time, config, service_ended) do
+    first_departure =
+      RealtimeSigns.headway_engine().get_first_scheduled_departure(
+        SourceConfig.sign_stop_ids(config)
+      )
+
+    last_scheduled_departure =
+      RealtimeSigns.headway_engine().get_last_scheduled_departure(
+        SourceConfig.sign_stop_ids(config)
+      )
+
+    last_actual_departure =
+      if service_ended do
+        SourceConfig.sign_stop_ids(config)
+        |> Stream.flat_map(&RealtimeSigns.last_trip_engine().get_recent_departures(&1))
+        |> Enum.max_by(fn {_key, datetime} -> datetime end, fn -> nil end)
+        |> case do
+          {_key, dt} -> dt
+          nil -> nil
+        end
```

Comment:
I think you could simplify this a bit by having the fallback just return {nil, nil} and then piping the result to an `elem(1)`

Reply:
nice 👍